packages/core/src/animation/AnimatorState.ts

```diff
@@ -0,0 +1,149 @@
+import { AnimationClip } from "./AnimationClip";
+import { Entity } from "../Entity";
+import { AnimatorStateTransition } from "./AnimatorTransition";
+import { WrapMode } from "./enums/WrapMode";
+import { PlayType } from "./enums/PlayType";
+import { AnimatorStateType } from "./enums/AnimatorStateType";
+
+export interface AnimatorStateMap {
+  [key: string]: AnimatorState;
+}
+
+export class AnimatorState {
+  /**
+   * The name mapping of all states.
+   */
+  static statesMap: AnimatorStateMap = {};
+  /**
+   * Get the state by name.
+   * @param name  The layer's name.
+   */
+  static findStateByName(name: string) {
+    return AnimatorState.statesMap[name];
+  }
+
+  /**
+   * The transitions that are going out of the state.
+   */
+  transitions: AnimatorStateTransition[] = [];
+  /**
+   * The speed of the clip. 1 is normal speed, default 1.
+   */
+  speed: number = 1;
+  /**
+   * The wrap mode used in the state.
+   */
+  wrapMode: WrapMode = WrapMode.LOOP;
+
+  /**
+   * Get the clip that is being played by this animator state.
+   */
+  get clip(): AnimationClip {
+    return this._clip;
+  }
+
+  /**
+   * Set the clip that is being played by this animator state.
+   */
+  set clip(clip: AnimationClip) {
+    if (this._target) {
+      clip._setTarget(this._target);
+    }
+    this._clip = clip;
+  }
+
+  /**
+   * Get the current time of the clip.
+   */
+  get frameTime(): number {
+    return this._frameTime;
+  }
+
+  /**
+   * Set the current time of the clip.
+   */
+  set frameTime(time: number) {
```

Comment:
I think this is internal. don't need to public for developer

Reply:
done.